File: Web/coffee/lib/service/order_service.dart
import 'package:coffee/model/order.dart';
import 'package:coffee/model/result.dart';
import 'package:coffee/model/user.dart';
import 'package:coffee/util/dio_app.dart';
import 'package:dio/dio.dart';

Future<Result<String>> pushOrderFinishNetwork(int orderId) async {
  Response response = await appDio
      .post("/order/pushOrderFinish", queryParameters: {"orderId": orderId});
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<String>> returnOrderNetwork(int orderId) async {
  Response response = await appDio.post("/order/returnOrder",
      queryParameters: {"orderId": orderId, "type": User.USER_MANAGER});
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<String>> deleteOrderNetwork(int orderId) async {
  Response response = await appDio
      .post("/order/deleteOrder", queryParameters: {"orderId": orderId});
  Result<String> result = Result.fromJson(response.data, (s) => s);
  return result;
}

Future<Result<List<Order>>> getOrderList() async =>
    await appDio.get("/order/orderList").then((value) => Result.fromJson(
        value.data,
        (orders) => (orders as List).map((e) => Order.fromJson(e)).toList()));

Future<Result<List<Order>>> getUserOrders(int userId) async => await appDio
    .get("/order/getUserOrders", queryParameters: {"userId": userId}).then(
        (value) => Result.fromJson(value.data,
            (json) => (json as List).map((e) => Order.fromJson(e)).toList()));

Future<Result<String>> receiveOrder(int orderId) async => await appDio
    .post("/order/receiveOrder", queryParameters: {"orderId": orderId}).then(
        (value) => Result.fromJson(value.data, (s) => s));

File: Server/src/main/java/com/drink/coffee/service/OrderService.java
package com.drink.coffee.service;

import com.drink.coffee.model.Result;
import com.drink.coffee.pojo.Order;

import java.util.List;

public interface OrderService {
    /**
     * 订单列表
     *
     * @return 订单列表
     */
    Result<List<Order>> orderList();

    /**
     * 下单
     *
     * @param order 订单
     * @return 创建结果
     */
    Result<String> createOrder(Order order);

    /**
     * 通知订单完成
     *
     * @param orderId 订单号
     * @return 通知结果
     */
    Result<String> pushOrderFinish(int orderId);

    /**
     * 退单
     *
     * @param orderId 订单号
     * @param type    退单人员
     * @return 退单结果
     */
    Result<String> returnOrder(int orderId, int type);

    /**
     * 删除订单
     *
     * @param orderId 订单号
     * @return 删除结果
     */
    Result<String> deleteOrder(int orderId);

    /**
     * 通过用户ID获取订单列表
     *
     * @param userId 用户Id
     * @return 订单列表
     */
    Result<List<Order>> getOrderListByUserId(int userId);

    /**
     * 接收订单
     *
     * @param orderId 订单Id
     * @return 接收结果
     */
    Result<String> receiveOrder(int orderId);
}
